### ISSUE TITLE
support customization of the viewer modes: single-panel view and multi-panel view

### ISSUE THREAD

**Issue by kswang1029** [idia-astro/carta-python#75]
as part of the epic https://github.com/CARTAvis/carta-python/issues/42, we will need a function to control the layout of the image viewer. Currently CARTA supports single-panel view and multi-panel view. There is a button at the top-right corner of the widget for switching the two modes. 
![Screen Shot 2023-03-07 at 19 08 59](https://user-images.githubusercontent.com/20819712/223405621-bac4f887-40ea-4ede-8021-8ff1e49450ab.png)

If it is the multi-panel view mode, and there are more images loaded (eg 5) than the configured grid layout (default 2x2), we can also use the adjacent buttons to switch to other "pages".

In the settings dialog, we can configure the grid layouts by setting n_row and n_column.

In short, we need a function or functions to configure the grid layout of the image viewer.

**Comment by Jordatious** [comment]
Agreed, configuring the multi-panel view would be good. To switch to single-panel view, a user should simply load in one image.

**Comment by I-Chenn** [comment]
@kswang1029 
I have made a quick test to this issue (switching between single and multiple panels) and was able to make it, though, by modifying the frontend code by adding an `@action` decorator prefix to `setImageMultiPanelEnabled` in `WidgetsStore` to get access of the function. And I guess to make actions of "next page" and "previous page" or "setting n_rows and n_columns", I also need to make similar frontend modification to at least `onNextPageClick` and `onPreviousPageClick` of `WidgetsStore`; `imagePanelColumns` and `imagePanelRows` of `PreferenceStore` (I haven't test them yet).

Should I address an issue to the frontend and make the these frontend modifications?

**Comment by kswang1029** [comment]
@I-Chenn thanks for investigating this. I suggest we make a frontend issue to address what need to be modified and assign yourself.

**Comment by I-Chenn** [comment]
@kswang1029 , @confluence 
I would like to look for solution to this issue.
As I mentioned above, I need to add `@action` decorator to `setImageMultiPanelEnabled`, `onNextPageClick` and `onPreviousPageClick` in `WidgetsStore.ts` to get access of using the functions in Python wrapper. I thus opened an issue (https://github.com/CARTAvis/carta-frontend/issues/2124) at frontend and did some tests and was able to make it work to take control of the three actions mentioned above.
Nevertheless, @YuHsuan-Hwang suggests that it's not logically proper to add the `@action` decorator to these functions considering the front-end use of them. I would like to know if there's an alternative way to call functions without `@action` decorator with Python wrapper. I did some surveys for the `request_scripting_action` of `protocol.py` but was not able to get a solution to the issue. May I have some tips (or any direction will do) for the issue?

**Comment by kswang1029** [comment]
I am not sure what the question is. It works if we do

session.call_action("widgetsStore.setImageMultiPanelEnabled", True)
session.call_action("widgetsStore.setImageMultiPanelEnabled", False)

then on the GUI we can see state of multi-panel changes accordingly.

**Comment by I-Chenn** [comment]
OK, thanks.
I didn't try that since I was told that the frontend functions to called needs to be an action. I should have at least tried that once. Thanks a lot.

**Comment by confluence** [comment]
@I-Chenn Sorry, I think I misled you  -- I also thought this, but apparently it's enough for the function to be exposed in the store even without this decorator.

**Comment by I-Chenn** [comment]
> @I-Chenn Sorry, I think I misled you -- I also thought this, but apparently it's enough for the function to be exposed in the store even without this decorator.

No problem at all : )
I should at least try once before asking you about that. Thanks a lot!
